areader/reader.go

```diff
@@ -340,8 +340,8 @@ func (ar *Reader) readHeaderV3(tReader *tar.Reader,
 	manifestChecksumStore := artifact.NewChecksumStore()
 	parsePath := []string{}
 
-	// Register the update handler for rootfsV3.
-	rootfs := handlers.NewRootfsInstaller(3)
+	// Register the update handler.
+	rootfs := handlers.NewRootfsInstaller()
```

Comment:
Sry for another bulk with a lot of files changing, but there really isn't much to it this time. 
1. I have changed the handler to not accept the version, and instead accept version as a parameter to the `handler.ReadHeader` function. This was required, as the client adds the install-handler before running `ReadArtifact`.